test/core/load_balancing/bm_picker.cc

```diff
@@ -0,0 +1,260 @@
+  };
+
+  class ChannelControl final
+      : public LoadBalancingPolicy::ChannelControlHelper {
+   public:
+    explicit ChannelControl(BenchmarkHelper* helper) : helper_(helper) {}
+
+    RefCountedPtr<SubchannelInterface> CreateSubchannel(
+        const grpc_resolved_address& address,
+        const ChannelArgs& per_address_args, const ChannelArgs& args) override {
+      return MakeRefCounted<SubchannelFake>(helper_);
+    }
+
+    void UpdateState(
+        grpc_connectivity_state state, const absl::Status& status,
+        RefCountedPtr<LoadBalancingPolicy::SubchannelPicker> picker) override {
+      MutexLock lock(&helper_->mu_);
+      helper_->picker_ = std::move(picker);
+      helper_->cv_.SignalAll();
+    }
+
+    void RequestReresolution() override { LOG(FATAL) << "unimplemented"; }
+
+    absl::string_view GetTarget() override { return "foo"; }
+
+    absl::string_view GetAuthority() override { return "foo"; }
+
+    RefCountedPtr<grpc_channel_credentials> GetChannelCredentials() override {
+      LOG(FATAL) << "unimplemented";
+    }
+
+    RefCountedPtr<grpc_channel_credentials> GetUnsafeChannelCredentials()
+        override {
+      LOG(FATAL) << "unimplemented";
+    }
+
+    grpc_event_engine::experimental::EventEngine* GetEventEngine() override {
+      return helper_->event_engine_.get();
+    }
+
+    GlobalStatsPluginRegistry::StatsPluginGroup& GetStatsPluginGroup()
+        override {
+      return helper_->stats_plugin_group_;
+    }
+
+    void AddTraceEvent(TraceSeverity severity,
+                       absl::string_view message) override {
+      LOG(FATAL) << "unimplemented";
+    }
+
+    BenchmarkHelper* helper_;
+  };
+
+  const absl::string_view name_;
+  const absl::string_view config_json_;
+  std::shared_ptr<grpc_event_engine::experimental::EventEngine> event_engine_ =
+      grpc_event_engine::experimental::GetDefaultEventEngine();
+  std::shared_ptr<WorkSerializer> work_serializer_ =
+      std::make_shared<WorkSerializer>(event_engine_);
+  OrphanablePtr<LoadBalancingPolicy> lb_policy_ =
+      CoreConfiguration::Get().lb_policy_registry().CreateLoadBalancingPolicy(
+          name_, LoadBalancingPolicy::Args{
+                     work_serializer_, std::make_unique<ChannelControl>(this),
+                     ChannelArgs()});
+  RefCountedPtr<LoadBalancingPolicy::Config> config_;
+  Mutex mu_;
+  CondVar cv_;
+  RefCountedPtr<LoadBalancingPolicy::SubchannelPicker> picker_
+      ABSL_GUARDED_BY(mu_);
+  absl::flat_hash_set<
+      std::shared_ptr<SubchannelInterface::ConnectivityStateWatcherInterface>>
+      connectivity_watchers_ ABSL_GUARDED_BY(mu_);
+  GlobalStatsPluginRegistry::StatsPluginGroup stats_plugin_group_ =
+      GlobalStatsPluginRegistry::GetStatsPluginsForChannel(
+          experimental::StatsPluginChannelScope("foo", "foo"));
+};
+
+void BM_Pick(benchmark::State& state, BenchmarkHelper& helper) {
+  helper.UpdateState(state.range(0));
+  auto picker = helper.GetPicker();
```

Comment:
If the purpose of this benchmark is to include just the picker, is there a way we can get the picker before we start the benchmark and pass it in?  Seems like there could be a lot of overhead here that could skew the numbers.

Reply:
the benchmarking lib doesn't start timing until the first iteration... there's no skew here